### PR TITLE
Add sticky sidebar to GoalPreview

### LINKED ISSUE #1748: Add sticky sidebar to GoalPreview
image

### PR DESCRIPTION
**Pull request by DenisVorop**
## PR includes
- [x] Feature

## Related issues
Resolve #1748 

## QA Instructions, Screenshots, Recordings
  

https://github.com/taskany-inc/issues/assets/95316053/948106f9-1f3c-4f26-b979-d623d7340849

### CODE REVIEW

File: src/components/GoalPreview/GoalPreview.tsx [hunk 1]
```diff
@@ -32,40 +39,82 @@ const StyledModalHeader = styled(ModalHeader)`
...
-                onClick: dispatchModalEvent(ModalEvent.GoalDeleteModal),
-            },
-        ],
-        [],
+            invalidate();
+        },
+        [goal, invalidate, goalTagsUpdate],
+    );
+
+    const onGoalAssigneeChange = useCallback(
```
Comment (awinogradov) [inline src/components/GoalPreview/GoalPreview.tsx:96]: This wrapper is not really needed

File: src/components/GoalPreview/GoalPreview.i18n/ru.json [hunk 2]
```diff
@@ -1,4 +1,4 @@
 {
-    "Edit": "Изменить",
+    "Edit": "Редактировать",
     "Delete": "Удалить"
```
Comment (awinogradov) [inline src/components/GoalPreview/GoalPreview.i18n/ru.json:3]: Unused key

File: src/components/GoalPreview/GoalPreview.tsx [hunk 3]
```diff
@@ -32,40 +39,82 @@ const StyledModalHeader = styled(ModalHeader)`
 `;
 
 const StyledModalContent = styled(ModalContent)`
-    overflow: auto;
+    z-index: 2;
```
Comment (awinogradov) [inline src/components/GoalPreview/GoalPreview.tsx:42]: ?

File: src/components/GoalPreview/GoalPreview.tsx [hunk 4]
```diff
@@ -32,40 +39,82 @@ const StyledModalHeader = styled(ModalHeader)`
...
+        goalProjectChange,
+        onGoalStateChange,
+        goalTagsUpdate,
+        goalOwnerUpdate,
+        invalidate,
+        onGoalParticipantAdd,
+        onGoalParticipantRemove,
+    } = useGoalResource({ id: goal?.id }, { invalidate: { getById: shortId } });
+
+    const onGoalTagAdd = useCallback(
```
Comment (awinogradov) [inline src/components/GoalPreview/GoalPreview.tsx:73]: Lets make issue about moving tag add/remove handlers to useGoalResource hook. They are fully copy and pasted here and on GoalPage.

**Review by awinogradov — CHANGES_REQUESTED** [review]

File: src/components/GoalPreview/GoalPreview.tsx (hunk 4 above)
Comment (DenisVorop) [reply src/components/GoalPreview/GoalPreview.tsx:74]: onGoalAssigneeChange is also a fully copy

https://github.com/taskany-inc/issues/issues/1853

Comment (awinogradov) [reply src/components/GoalPreview/GoalPreview.tsx:74]: onGoalAssigneeChange removed here https://github.com/taskany-inc/issues/pull/1824

**Review by awinogradov — APPROVED** [review]

**Review by 9teen90nine — APPROVED** [review]

**Pull request merged**